### ISSUE TITLE
2.2.2 sets the `@JsonProperty` when it is not needed.

### ISSUE THREAD

**Issue by ldriscoll** [immutables/immutables#353]
for our json serialisation, in certain circumstances, we set a Jackson Naming Strategy of `LowerCaseWithUnderscores`; even if this is set, if a method is annotated with a value in it's `@JsonProperty` this overwrites the naming strategy.  Now since 2.2.1 we have a problem:
```java
@Value.Immutable
interface Foo {
  String getExternalIdentifier();
}
```
used to serialize as `{"external_identifier": "bar"}`, since the generated `ImmutableFoo` would be:
```
@JsonProperty
public String getExternalIdentifier() {
 ...
}
```

However, since 2.2.1 it generates
```
@JsonProperty("externalIdentifier")
public String getExternalIdentifier() {
 ...
}
```
So the serialized json is *ALWAYS* `{"externalIdentifier": "bar"}`,  I believe if people want to have non standard serialized values then they should annotate the methods themselves.  I believe the change was made so that it would allow for immutables where people didn't use the standard `getX` and `isX`

**Comment by ldriscoll** [comment]
just to add, this absolutely prevents us from upgrading to 2.2.2.

**Comment by elucash** [comment]
With that change we painted ourselves in the corner even more, but frankly, I'm not ready to reverse that. I've tried couple of approaches how to mitigate that, but unfortunately, the only working solution I've found (with a tradeoff how many people may use naming strategy vs how many people might use Immutables without configuring "is*" for example) is to have a knob to disable forcing names, but which is enabled by default. In your case you'll need to set `Style(forceJacksonPropertyNames = false)` in order for the field naming strategies to work. I'm still looking for a better solution, but given how brittle things become, that better solution need to be damn good.
The released version is 2.2.3 to try it out
